File: src/vlp_fir/src/Marker.cpp
//
// Created by little-bird on 2019/10/31.
//

#include <Marker.h>

using namespace std;
vlp_fir::send_msgs obj1;
vlp_fir::send_msgs marker_(vector<pcl::PointXYZ> &save_aabb_max,vector<pcl::PointXYZ> &save_aabb_min,visualization_msgs::Marker &marker,int n,int i){
    //marker.header.frame_id = "velodyne";
    marker.header.frame_id = "lidar";
    marker.header.stamp = ros::Time();
    marker.ns = "my_namespace";
    marker.id = n;
    marker.type = visualization_msgs::Marker::CUBE;
    marker.action = visualization_msgs::Marker::ADD;

    marker.pose.position.x = (save_aabb_max[i].x + save_aabb_min[i].x)/2;
    marker.pose.position.y = (save_aabb_max[i].y + save_aabb_min[i].y)/2;
    marker.pose.position.z = (save_aabb_max[i].z + save_aabb_min[i].z)/2;

    cout<<"x "<< marker.pose.position.x<<"  y "<<marker.pose.position.y<<endl;
    marker.pose.orientation.x = 0.0;
    marker.pose.orientation.y = 0.0;
    marker.pose.orientation.z = 0.0;
    marker.pose.orientation.w = 1.0;

    marker.scale.x = save_aabb_max[i].x - save_aabb_min[i].x;
    marker.scale.y = save_aabb_max[i].y - save_aabb_min[i].y;
    marker.scale.z = save_aabb_max[i].z - save_aabb_min[i].z;

    marker.color.a = 1; // Don't forget to set the alpha!
    marker.color.r = 0.0;
    marker.color.g = 1.0;
    marker.color.b = 0.0;
    marker.lifetime = ros::Duration(0.1);
    cout<<"L:"<<marker.scale.x<<"  W:"<<marker.scale.y<<" H:"<<marker.scale.z<<endl;
    if(save_aabb_max.size()>0) {
        obj1.num = save_aabb_max.size();
        obj1.x = marker.pose.position.x;
        obj1.y = marker.pose.position.y;
        obj1.z = marker.pose.position.z;
        obj1.len = marker.scale.x;
        obj1.width = marker.scale.y;
        obj1.height = marker.scale.z;
        obj1.date = 1;

    }
    else {
        obj1.date = 0;
        obj1.num = 0;
    }
    return obj1;

}

File: scout/src/multi_goal/Marker.py
import rospy
from visualization_msgs.msg import Marker
from std_msgs.msg import Int16MultiArray

if __name__ == '__main__':
    x = 0
    y = 0
    rospy.init_node('goal_Marker')
    while not rospy.is_shutdown():

        data = rospy.rospy.wait_for_message('goal_rand', Int16MultiArray)
        if len(data):
            x = data.data[0]
            y = data.data[1]

        goal_marker = rospy.Publisher('goal_Marker', Marker, queue_size = 10)

        pub_goal_msg = Marker()
        pub_goal_msg.ns = "goal_Marker"
        pub_goal_msg.header.frame_id = "/odom"
        pub_goal_msg.header.stamp = rospy.Time.now()
        pub_goal_msg.id = 0
        pub_goal_msg.pose.position.x = x
        pub_goal_msg.pose.position.y = y
        pub_goal_msg.lifetime = rospy.Duration(0)
        pub_goal_msg.pose.position.z = 1
        pub_goal_msg.pose.orientation.x = 0.0
        pub_goal_msg.pose.orientation.y = 0.0
        pub_goal_msg.pose.orientation.z = 0.0
        pub_goal_msg.pose.orientation.w = 1.0
        pub_goal_msg.scale.x = 0.5
        pub_goal_msg.scale.y = 0.5
        pub_goal_msg.scale.z = 0.03
        pub_goal_msg.color.a = 0.3
        pub_goal_msg.color.r = 1.0
        pub_goal_msg.color.g = 0.0
        pub_goal_msg.color.b = 0.0

        goal_marker.publish(pub_goal_msg)